### PR TITLE
Small refactor in datapath/linux/node.go

### PR DESCRIPTION
**Pull request by derailed**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

This PR addresses a few small clean up items in datapath/linux/node.go

- Fix typos
- Dry up valid IP check
- Ensure errors are propagated

Fixes: #issue-number

Signed-off-by: Fernand Galiana <fernand.galiana@isovalent.com>

### CODE REVIEW

**Review by gentoo-root — APPROVED** [review]

**Comment by julianwiedmann** [comment]
Let's give https://github.com/cilium/cilium/pull/28941 a chance to land first, in case there's any conflict.

**Comment by julianwiedmann** [comment]
@derailed fyi, Paul's PR is in. And as expected, this one picked up some conflict :boom:.

**Review by gentoo-root — COMMENTED** [review]
The changes mentioned in the commit message seem no longer here:

> - Fix typo
> - Dry up valid IP check
> - Ensure errors are propagated

File: pkg/datapath/linux/ipsec.go [hunk 1]
```diff
@@ -98,8 +98,10 @@ func (n *linuxNodeHandler) enableIPsec(newNode *nodeTypes.Node, nodeID uint16) e
 }
 
 func (n *linuxNodeHandler) enableIPsecIPv4(newNode *nodeTypes.Node, nodeID uint16, zeroMark bool) error {
-	var spi uint8
-	var errs error
+	var (
```
Comment (gentoo-root) [inline pkg/datapath/linux/ipsec.go]: Does the syntax with a single var block have any benefits?

IMO, unrelated styling changes should be avoided if both styles are permitted, because:

1. It makes `git blame` less useful.
2. It could make backporting and reverting more troublesome.
3. It adds some noise that interferes with reviewing.

**Review by gentoo-root — APPROVED** [review]

File: pkg/datapath/linux/ipsec.go (hunk 1 above)
Comment (derailed) [reply pkg/datapath/linux/ipsec.go]: No benefits besides DRY, line of sight and declaring vars where they are used.
I'll remove it.

**Review by pchaigno — CHANGES_REQUESTED** [review]
Thanks for the pull request!

It seems that it fixes a case where an error isn't reported. That's a fix and should be in its own pull request that we can separately backport.

In addition, it would be good to split the changes in multiple commits with proper descriptions as this is going to cause rebase & backport conflicts for other works; having a proper commit history helps resolution of those conflicts.

**Comment by derailed** [comment]
@pchaigno Thanks for the feedback! 
As I enter the seventh layer of backporting hell, I can certainly sympathise...
Updated this PR per you suggestions and spin off (https://github.com/cilium/cilium/pull/29423)

File: pkg/datapath/linux/node.go [hunk 2]
```diff
@@ -1213,8 +1213,7 @@ func (n *linuxNodeHandler) NodeConfigurationChanged(newConfig datapath.LocalNode
 				return fmt.Errorf("direct routing device is required, but not defined")
 			}
 
-			var targetDevices []string
-			targetDevices = append(targetDevices, option.Config.DirectRoutingDevice)
+			targetDevices := []string{option.Config.DirectRoutingDevice}
```
Comment (pchaigno) [inline pkg/datapath/linux/node.go]: Any reason not to use `make` with the length of `option.Config.GetDevices()` + 1 here? (Note I'm a Go newbie.)

File: pkg/datapath/linux/node.go [hunk 3]
```diff
@@ -1401,8 +1399,10 @@ func (n *linuxNodeHandler) NodeCleanNeighborsLink(l netlink.Link, migrateOnly bo
 		defer n.neighLock.Unlock()
 	}
 
-	var neighSucceeded, neighErrored int
-	var which string
+	var (
+		neighSucceeded, neighErrored int
+		which                        string
+	)
```
Comment (pchaigno) [inline pkg/datapath/linux/node.go]: I'd avoid making changes like this and the one on line 1287. They are style preferences for which the Cilium project doesn't have rules AFAIK. They don't fix anything but can cause conflicts on backports & co. IMO, the drawbacks outweigh the benefits.

**Review by pchaigno — APPROVED** [review]
Thanks for reworking the commit history!

File: pkg/datapath/linux/node.go (hunk 2 above)
Comment (derailed) [reply pkg/datapath/linux/node.go]: Good catch!

File: pkg/datapath/linux/node.go (hunk 3 above)
Comment (derailed) [reply pkg/datapath/linux/node.go]: I agree besides DRY and readability no value add here. I'll revert it

**Comment by joestringer** [comment]
Heads up, I dropped the `release-note` for this to `misc` because refactors do not have user-facing impact. This is purely for ordering the release notes to help users more easily understand changes that may impact them, it has no relation to the value of the contribution. Thank you again for contributing during this release cycle.
